src/ffi/mod.rs

```diff
@@ -831,6 +844,72 @@ extern "C" {
     fn fortify_source();
 }
 
+#[cfg(feature = "hide_ips")]
+struct LogCensor {
+    ip_mask_seed: [u8; 32],
+    ip_masks: parking_lot::Mutex<HashMap<Ipv4Addr, String>>,
+    ip_regex: Regex,
+}
+
+#[cfg(feature = "hide_ips")]
+impl LogCensor {
+    fn new() -> Option<Self> {
+        Regex::new(
+            r"(?x)\b
+                (?:
+                    (?:
+                        25[0-5]
+                        |  2[0-4][0-9]
+                        |  1[0-9]{2}
+                        |  [1-9]?[0-9]
+                    )
+                    \.
+                ){3}
+                (?:
+                    25[0-5]
+                    |  2[0-4][0-9]
+                    |  1[0-9]{2}
+                    |  [1-9]?[0-9]
+                )
+                \b",
+        )
+        .ok()
+        .map(|re| LogCensor {
+            ip_masks: parking_lot::Mutex::new(HashMap::new()),
+            ip_mask_seed: rand::thread_rng().gen::<[u8; 32]>(),
+            ip_regex: re,
+        })
+    }
+
+    fn censor_logs(&self, log: String) -> String {
+        // Gather all of the IPs
+        self.ip_regex
+            .replace_all(&log, |captures: &Captures| {
+                // First capture is guaranteed to be present by regex crate
+                #[allow(index_access_check)]
+                let ip_match = &captures[0];
+                Ipv4Addr::from_str(ip_match)
+                    .map(|ip_addr| {
+                        self.ip_masks
+                            .lock()
+                            .entry(ip_addr)
+                            .or_insert_with(|| {
+                                // Probably good enough protection for this usecase
```

Comment:
So we have already had a place there in anlytics we are hasing server ip's with md5, i feel it would be nice to keep this consistency.

Reply:
https://github.com/NordSecurity/libtelio/blob/main/crates/telio-nurse/src/heartbeat.rs#L518

Reply:
Changed the hash algorithm to md5 to be more compliant, but left some additional noise to make it harder to reverse - is it ok @packgron

Reply:
I feel that for ipv4 it's never to to simply brute force all ip's. I think what you did is more than enough.